core_backend/tests/api/test_import_content.py

```diff
@@ -0,0 +1,265 @@
+    client: TestClient,
+    fullaccess_token: str,
+) -> Generator[str, None, None]:
+    response = client.post(
+        "/content",
+        headers={"Authorization": f"Bearer {fullaccess_token}"},
+        json={
+            "content_title": "Title in DB",
+            "content_text": "Text in DB",
+            "content_language": "ENGLISH",
+            "content_tags": [],
+            "content_metadata": {},
+        },
+    )
+    content_id = response.json()["content_id"]
+    yield content_id
+    client.delete(
+        f"/content/{content_id}",
+        headers={"Authorization": f"Bearer {fullaccess_token}"},
+    )
+
+
+def _dict_to_csv_bytes(data: dict) -> BytesIO:
+    """
+    Convert a dictionary to a CSV file in bytes
+    """
+
+    df = pd.DataFrame(data)
+    csv_bytes = BytesIO()
+    df.to_csv(csv_bytes, index=False)
+    csv_bytes.seek(0)
+
+    return csv_bytes
+
+
+@pytest.fixture
+def data_valid() -> BytesIO:
+    data = {
+        "content_title": ["csv title 1", "csv title 2"],
+        "content_text": ["csv text 1", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_empty_csv() -> BytesIO:
+    data: dict = {}
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_no_rows() -> BytesIO:
+    data: dict = {
+        "content_title": [],
+        "content_text": [],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_missing_columns() -> BytesIO:
+    data = {
+        "wrong_column_1": ["Value 1", "Value 2"],
+        "wrong_column_2": ["Value 3", "Value 4"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_title_missing() -> BytesIO:
+    data = {
+        "content_title": ["", "csv text 1"],
+        "content_text": ["csv title 2", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_text_missing() -> BytesIO:
+    data = {
+        "content_title": ["csv title 1", "csv title 2"],
+        "content_text": ["", "csv text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_long_title() -> BytesIO:
+    data = {
+        "content_title": ["a" * 151],
+        "content_text": ["Valid text"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_long_text() -> BytesIO:
+    data = {
+        "content_title": ["Valid title"],
+        "content_text": ["a" * 2001],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_duplicate_titles() -> BytesIO:
+    data = {
+        "content_title": ["Duplicate title", "Duplicate title"],
+        "content_text": ["Text 1", "Text 2"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_duplicate_texts() -> BytesIO:
+    data = {
+        "content_title": ["Title 1", "Title 2"],
+        "content_text": ["Duplicate text", "Duplicate text"],
+    }
+    return _dict_to_csv_bytes(data)
+
+
+@pytest.fixture
+def data_title_in_db() -> BytesIO:
```

Comment:
Maybe test behaviour when there is a mismatch between the  number of titles and the number of texts

Reply:
That'd result in a parsing error I think - I should probably try loading some broken CSVs to test that...